tf2/include/tf2/time_cache.h

```diff
@@ -118,7 +118,7 @@ class TimeCache : public TimeCacheInterface
   L_TransformStorage storage_;
 
   ros::Duration max_storage_time_;
-
+  ros::Time last_repeated_warn_time_;
```

Comment:
This most probably wouldn't make it into Noetic. It changes the ABI of existing code, which would require a total rebuild of the whole Noetic buildfarm.

I suggest moving this completely to the .cpp . You can make this a static variable in the .cpp file, which would preserve ABI. It could also simplify the implementation.

If you no longer feel the urge to get this fix merged, ping @klaxalk , he told he could find some time to dedicate to this fix ;)